ToolBox/CoinApi/CoinApiDataQueueHandler.cs

```diff
@@ -326,5 +341,94 @@ private void OnError(object sender, Exception e)
         {
             Log.Error(e);
         }
+
+        #region SynchronizingHistoryProvider
+
+        public override void Initialize(HistoryProviderInitializeParameters parameters)
+        {
+            // NOP
+        }
+
+        public override IEnumerable<Slice> GetHistory(IEnumerable<HistoryRequest> requests, DateTimeZone sliceTimeZone)
+        {
+            var subscriptions = new List<Subscription>();
+            foreach (var request in requests)
+            {
+                var history = GetHistory(request);
+                var subscription = CreateSubscription(request, history);
+                subscriptions.Add(subscription);
+            }
+            return CreateSliceEnumerableFromSubscriptions(subscriptions, sliceTimeZone);
+        }
+
+        public IEnumerable<BaseData> GetHistory(HistoryRequest historyRequest)
+        {
+            if (historyRequest.Symbol.SecurityType != SecurityType.Crypto)
+            {
+                Log.Error($"CoinApiDataQueueHandler.GetHistory(): Invalid security type {historyRequest.Symbol.SecurityType}");
+                yield break;
+            }
+
+            if (historyRequest.Resolution == Resolution.Tick)
+            {
+                Log.Error("CoinApiDataQueueHandler.GetHistory(): Historical ticks can not be provided, only OHLCV timeseries");
+                yield break;
+            }
+
```

Comment:
Should also add a check for the tick type, since current implementation only supports Trade information, not Quote